packages/protocol/src/utils/SPID.ts

```diff
@@ -0,0 +1,202 @@
+import { format } from 'util'
+
+/**
+ * Object version of SPID
+ */
+export type SPIDObject = {
+    streamId: string,
+    streamPartition: number
+}
+
+/**
+ * Represents partial SPIDObject e.g. for setting defaults
+ * Known keys
+ */
+export type SPIDObjectPartial = Partial<SPIDObject>
+
+/**
+ * SPID or String representing a SPID
+ * Object cases can be typechecked
+ * TODO: SPID string type safety
+ */
+export type SPIDLike = SPID | string | SPIDObject | { id: string, partition: number }
+
+/**
+ * Flexible input type
+ */
+export type SPIDLikePartial = SPIDLike | SPIDObjectPartial | Partial<{ id: string, partition: number }>
```

Comment:
This definition allows `{ partition: 123 }` to be `SPIDLikePartial`? Intentional?

Reply:
Intentional